### PR TITLE
Add initial kotlin setuppayload implementation

### PR DESCRIPTION
**Pull request by yufengwangca**
This is the initial setuppayload implementation in kotlin, it still rely on the existing native setup_payload implementation, the final goal is to implement setup payload completely in kotlin and deprecate the existing src/setup_payload implemented in java/c++.

Testing:
Leverage the existing CI tests using java_matter_controller

### CODE REVIEW

File: src/controller/java/src/chip/setuppayload/SetupPayloadParser.kt [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  ): SetupPayload
+
+  class UnrecognizedQrCodeException(qrCode: String) :
+    Exception(String.format("Invalid QR code string: %s", qrCode), null) {
+    companion object {
+      private const val serialVersionUID = 1L
+    }
+  }
+
+  class InvalidEntryCodeFormatException(entryCode: String) :
```
Comment (tcarmelveilleux) [inline src/controller/java/src/chip/setuppayload/SetupPayloadParser.kt:83]: "Manual pairing code" is the name of in the spec. "Entry code" is new terminology.

File: src/controller/java/src/chip/setuppayload/SetupPayloadParser.kt [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  @Throws(UnrecognizedQrCodeException::class, SetupPayloadException::class)
+  fun parseQrCode(qrCodeString: String): SetupPayload {
+    return fetchPayloadFromQrCode(qrCodeString, false)
+  }
+
+  /**
+   * Returns [SetupPayload] parsed from the QR code string.
+   *
+   * @param qrCodeString the QRCode for commissioning device.
+   * @param allowInvalidPayload Allow invalid payload components. If this value is true, payload
```
Comment (tcarmelveilleux) [inline src/controller/java/src/chip/setuppayload/SetupPayloadParser.kt:22]: Why would we allow invalid payloads?

File: src/controller/java/src/chip/setuppayload/SetupPayloadParser.kt [hunk 3]
```diff
@@ -0,0 +1,108 @@
+package chip.setuppayload
+
+import java.util.logging.Level
+import java.util.logging.Logger
+
+/** Parser for scanned QR code or manual entry code.  */
+class SetupPayloadParser {
```
Comment (tcarmelveilleux) [inline src/controller/java/src/chip/setuppayload/SetupPayloadParser.kt:7]: In spec it is called "onboarding payload".  "Setup Payload" is the legacy pre-spec SDK name.

File: src/controller/java/src/chip/setuppayload/SetupPayloadParser.kt [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  }
+
+  /**
+   * Returns [SetupPayload] parsed from the manual entry code string. If an SetupPINCode has
+   * invalid value, SetupPayloadException occurs. Refer to [SetupPayload] for the description
+   * of the invalid element.
+   */
+  @Throws(InvalidEntryCodeFormatException::class, SetupPayloadException::class)
+  fun parseManualEntryCode(entryCodeString: String): SetupPayload {
+    return fetchPayloadFromManualEntryCode(entryCodeString, false)
```
Comment (tcarmelveilleux) [inline src/controller/java/src/chip/setuppayload/SetupPayloadParser.kt:40]: "fetch" as a verb implies going somewhere. This is just a conversion/parsing.

Curious why it is called "fetchXxxx"

File: src/controller/java/src/chip/setuppayload/SetupPayloadParser.kt (hunk 2 above)
Comment (andy31415) [reply src/controller/java/src/chip/setuppayload/SetupPayloadParser.kt]: We are porting existing code, not re-making it.

#22357 seems to have added it, I am guessing it is for some tests that are run and you must accept invalid data somehow. Changing tests is outside of the scope of a port. Possible and desirable I believe, however as a separate task (and I would suggest after things are moved to use less C++).

Comment (andy31415) [reply src/controller/java/src/chip/setuppayload/SetupPayloadParser.kt]: The name is probably bad .... skipContentValidation is probably a better name based on the comments in the original PR.

Comment (yufengwangca) [reply src/controller/java/src/chip/setuppayload/SetupPayloadParser.kt]: Renamed to skipPayloadValidation

**Review by joonhaengHeo — APPROVED** [review]

**Review by joonhaengHeo — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**